CMakeLists.txt

```diff
@@ -1,26 +1,28 @@
-cmake_minimum_required(VERSION 3.1.0 FATAL_ERROR)
+if(TARGET mapbox-base)
+    return()
+endif()
+
+cmake_minimum_required(VERSION 3.5.1 FATAL_ERROR)
 project(MAPBOX_BASE LANGUAGES CXX C)
 
-enable_testing()
+include(CTest)
 
-# Required to avoid warnings when setting visibility
 cmake_policy(SET CMP0063 NEW)
 
+if(NOT CMAKE_VERSION VERSION_LESS 3.13.0)
+    # Required to allow mapbox-base to depend on libraries coming from /deps
+    cmake_policy(SET CMP0079 NEW)
+endif()
+
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
 
-set (MAPBOX_BASE_EXTRAS ${CMAKE_CURRENT_LIST_DIR}/extras)
+add_subdirectory(${PROJECT_SOURCE_DIR}/extras)
+add_subdirectory(${PROJECT_SOURCE_DIR}/include)
+add_subdirectory(${PROJECT_SOURCE_DIR}/deps)
 
-if(TARGET mapbox-base)
-    return()
+if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING)
```

Comment:
```cpp
CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME
```

Why we need such check? We can just setup `BUILD_TESTING` variable in project that include this cmake. 
Your code don't allow developer to control whether it want to build tests

Reply:
We need this check to avoid building tests for `mapbox-base` when this is used as dependency for other projects. This is part of the recommended practiced for Modern CMake usage ref: https://cliutils.gitlab.io/modern-cmake/chapters/testing.html

Reply:
But why you decided that our team doesn't need that?
How we can check all libraries on our target device in TME project?

Reply:
Running `mapbox-base` tests outside of the project is redundant: the CI checks we make guarantees that every code commited to `master` is tested and validated. If yet you still need these, we can add them as a follow-up PR.

Reply:
We want to run automated checks for everything we have in our project before deliver a product. We want to run it on a target device. It is not possible to do that with CI (in opensource project). And why testing of one product for specified case should be in a common repo?

Reply:
@deniskoronchik let's add support for running tests back on a follow-up PR :+1: